Jenkinsfile

```diff
@@ -11,7 +12,23 @@ pipeline {
     ansiColor 'xterm'
   }
 
+  triggers {
+    issueCommentTrigger(TRIGGER_PATTERN)
+  }
+
   stages {
+    stage('Validate PR Source') {
+      when {
+        expression { env.CHANGE_FORK }
+        not {
+          triggeredBy 'issueCommentCause'
+        }
+      }
+      steps {
+        error("A maintainer needs to approve this PR for CI by commenting")
+      }
+    }
+
```

Comment:
Wonder if it is possible to move this into the magic butter libs so we don't have to keep coping this + the regex around.

A function that returns a stage.

Reply:
looking into this separately from these PRs, I think it'd be good to have more of these identical steps broken out into magical butler abstractions

Reply:
I hereby submit a petition to rename the jenkins library `magic-butter` as a much more worthy name.